### ISSUE TITLE
Bigquery dialect: Rules L029 and L059 are contradicting each other

### ISSUE THREAD

**Issue by michalklempa** [sqlfluff/sqlfluff#4480]
### Search before asking

- [X] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

If one desperately needs keyword as a column name, sqlfluff can't be satisfied as at any time, either rule L029  or L059 is violated.

### Expected Behaviour

Let me suggest two changes:

1. rule `L029` error message would be extended to:

```
Keywords should not be used as identifiers. If unavoidable, quote the the identifier
```

2. rule `L059` behaviour would be changed to consider quoted identifier as necessary, if the word being quoted is in keywords set.

### Observed Behaviour

Example SQL in BigQuery dialect:

```sql
CREATE TABLE ordered_students_by_index (
index INT64,
)
```

Using 

```
sqlfluff lint --dialect bigquery <file>
```

We get:

```
L:   2 | P:   3 | L029 | Keywords should not be used as identifiers.
```

So far so good, but sometimes one can't avoid having a keyword as an identifier, so there, quoting comes into play.

After changing the SQL to:

```sql
CREATE TABLE ordered_students_by_index (
`index` INT64,
)
```

And re-running sqlfluff, we get:

```
L:   2 | P:   3 | L059 | Unnecessary quoted identifier `index`.
```

Both rules make perfect sense, just sometimes it may be needed to just use the keyword as identifier and quote it properly.
 

### How to reproduce

.

### Dialect

bigquery

### Version

```
> sqlfluff --version
sqlfluff, version 1.4.5
```

```
> python --version
Python 3.8.6
```

### Configuration

default configuration

### Are you willing to work on and submit a PR to address the issue?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)

**Comment by alanmcruickshank** [comment]
@michalklempa I would suggest in this case instead to ignore that specific line. Where it's possible to avoid using keywords, I think they should be flagged whether or not they're quoted. I'm also hesitant to bind those two rules together more tightly.

This is exactly what the `--noqa` feature is for: https://docs.sqlfluff.com/en/stable/configuration.html#ignoring-individual-lines

There will always be times when we have to do things which aren't typically advisable, but by using a `noqa`, others contributing to your codebase will see that you _know_ that it's not normally a good idea to use keywords, but in this case there's a specific reason that it's necessary. That's much more explicit that hiding that functionality in the interaction between two rules.

**Comment by michalklempa** [comment]
And what is your opinion on possibly just loosen the L059 rule, so that: if unnecessary quoted identifier is actually a keyword, we would emit just warning instead of error?

I agree the L029 should not nudge people into "just quoting" identifiers which are actually keywords. Anyway, it seems that even if we would loosen the rules a bit, one would always need to put some `--noqa` somewhere. So as you think is it appropriate.

Thank you for your feedback.
